docs/learn/resource-materials/glossary.md

```diff
@@ -1,34 +1,196 @@
+- **Epoch (IOTA 2.0):** A time interval used for a specific type of consensus mana. At the end of each epoch, a snapshot is taken of the state of mana distribution on the network. Since this tool uses the timestamp of messages, each node can eventually reach a consensus on the mana distribution of an epoch.
+
+## F
+
+- **Faucet:** A pool of tokens (funds). Upon uncomplicated request, one gets a limited number of tokens for testing, especially for developers of own apps this is a great help.
+- **Firefly:** Firefly is a wallet, intended to serve as a platform for the current and future IOTA ecosystem.
+- **Finality:** The property that once a transaction has been completed, there is no way to reverse or change it. This is the moment when the parties involved in a transfer can consider the transaction completed. Finality can be deterministic or probabilistic.
+- **Full nodes (Hornet, Bee):** They form the core (infrastructure) of the IOTA network. In order to participate in the peer-to-peer network, the full node must always be online and connected to neighbors (other full nodes). In addition, the transaction database must be synchronized with all other full nodes in the network. The role of full nodes is to interact with clients (wallets, DApps, etc.) and attach their transactions to the ledger, make transactions known to all other full nodes in the network, validate transactions and store them in the ledger.
+- **Future Cone:** All messages that directly or indirectly reference a message are called its future cone.
+- **Fork:** In IT, this is a new development branch after a project is split into a second follow-on project; the source code or parts of it are developed independently of the original parent project.
+- **FPC(Fast Probabilistic Consensus):** Consensus that uses a random number and node opinions to reach consensus. In On-Tangle Voting, it is only used in a specific edge case. Check out OTVFPCS.
+
+## G
+
+- **Genesis transaction:** The Genesis transaction is the first transaction that created all IOTA tokens and distributed them to the addresses of the buyers.
+- **GoShimmer (No Main net):** Prototype of the coordinatorless version of IOTA in the Go programming language. GoShimmer implements the various modules of Coordicide, such as auto peering, node identities, Mana, etc. GoShimmer serves as a test environment for the first alpha version and the test network. Everything tested here will be gradually merged with Hornet and Bee.
+- **Generic Data Object (IOTA 2.0):** The most basic object type. All unrecognized data objects are treated this way.
+
+## H
+
+- **History:** The list of transactions that were directly or indirectly authorized by a particular transaction.
```

Comment:
Is it this kind of history asked here? It can also mean my transaction history etc... we should not try to define such broad words. better give a definition for transaction history